geomstats/learning/riemannian_mean_shift.py

```diff
@@ -1,147 +1,161 @@
 
-        for i in range(max_iter):
-            dists = self.__intersets_distances(
-                centers, points, n_jobs=n_jobs, **joblib_kwargs
-            )
+        print(type(self.max_iter))
+        print(self.max_iter)
+        for _ in range(self.max_iter):
+            dists = self.__intersets_distances(centers, X)
 
             # assuming the use of 'flat' kernel
-            if kernel == "flat":
-                weights = np.ones_like(dists)
+            if self.kernel == "flat":
+                weights = gs.ones_like(dists)
 
             weights[dists > self.bandwidth] = 0.0
-            weights = weights / weights.sum(axis=1, keepdims=1)
+            weights = weights / gs.sum(weights, axis=1, keepdims=1)
 
             points_to_average, nonzero_weights = [], []
 
-            for j in range(n_centers):
-                points_to_average += [
-                    points[np.where(weights[j, :] > 0)[0], :],
-                ]
+            for j in range(self.n_centers):
+                indexes = gs.where(weights[j] > 0)
                 nonzero_weights += [
-                    weights[j, :].nonzero()[0],
+                    weights[j][indexes],
+                ]
+                points_to_average += [
+                    X[indexes],
                 ]
 
             # compute Frechet means in parallel
-            pool = joblib.Parallel(n_jobs=n_jobs, **joblib_kwargs)
+            pool = joblib.Parallel(n_jobs=self.n_jobs)
             out = pool(
                 pickable_mean(points_to_average[j], nonzero_weights[j])
-                for j in range(centers.shape[0])
+                for j in range(self.n_centers)
             )
 
-            new_centers = np.array(out)
+            new_centers = gs.array(out)
 
-            displacements = [
-                self.metric.dist(centers[j], new_centers[j]) for j in range(n_centers)
-            ]
+            displacements = [self.metric.dist(centers[j], new_centers[j])]
             centers = new_centers
 
-            if (np.array(displacements) < self.tol).all():
+            if (gs.array(displacements) < self.tol).all():
                 break
 
         self.centers = centers
 
     def predict(self, points):
-        """
-        Predict the closest cluster each point in 'points' belongs to
-        """
-
+        """Predict the closest cluster each point in 'points' belongs to"""
         if self.centers is None:
             raise Exception("Not fitted")
-        else:
-            out = []
-            for i in range(points.shape[0]):
-                j_closest_center = self.metric.closest_neighbor_index(
-                    points[i, :], self.centers
-                )
-                out.append(self.centers[j_closest_center, :])
-
-            return np.array(out)
+
+        out = []
+        for i in range(points.shape[0]):
+            j_closest_center = self.metric.closest_neighbor_index(
+                points[i, :], self.centers
```

Comment:
use `points[i]` as points may be matrices.

Reply:
^This would become `point` with the syntax for point in points